addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java

```diff
@@ -142,20 +155,25 @@ protected void applyParameters(
             if (param.getValue() == null) {
                 continue;
             }
+
             String resolvedValue;
             if (env != null) {
                 resolvedValue = env.replaceVars(param.getValue());
             } else {
                 resolvedValue = param.getValue().toString();
             }
-            if (applyCustomParameter(key, resolvedValue)) {
-                progress.info(
-                        Constant.messages.getString(
-                                "automation.info.setparam",
-                                this.getType(),
-                                key,
-                                param.getValue().toString()));
-                continue;
+            if (!verify) {
+                if (applyCustomParameter(key, resolvedValue)) {
+                    progress.info(
+                            Constant.messages.getString(
+                                    "automation.info.setparam",
+                                    this.getType(),
+                                    key,
+                                    param.getValue().toString()));
+                    continue;
+                }
+            } else {
+                applyCustomParameter(key, resolvedValue, progress);
```

Comment:
Why are we applying the parameter if verify is true? That will change the state.
Say theres a job `X` with param `Y` which defaults to `true`.
The user might define job `X` twice, the first time assuming `Y` is `true` and the second time setting it to `false`.
If we apply the params during the initial verify step then `Y` will incorrectly end up being `false` before the first job is run.

Reply:
Why not do all of the validation in verifyJobSpecificData including the parameters? Could have a verifyCustomParameter method as well for convenience?
I'm still not really keep in the `verify` parameter on applyParameters.
Any thoughts on this @thc202 ?

Reply:
In this case it should be fine since this method is an implementation detail (it does need to be changed to package), but agree about the actual validation for others, it's preferable to have specific method e.g. `verifyCustomParameter` (even if internally it uses the same method to apply and verify, that should be an implementation detail).